sui-execution/latest/sui-adapter/src/execution_engine.rs

```diff
@@ -220,100 +220,85 @@ fn execute_transaction<Mode: ExecutionMode>(
-                    effects_estimated_size = effects_estimated_size,
-                    soft_limit = limit,
-                    "Estimated transaction effects size crossed soft limit",
-                )
-            }
-            LimitThresholdCrossed::Hard(_, lim) => {
-                execution_result = Err(ExecutionError::new_with_source(
-                    ExecutionErrorKind::EffectsTooLarge {
-                        current_size: effects_estimated_size as u64,
-                        max_size: lim as u64,
-                    },
-                    "Transaction effects are too large",
-                ))
-            }
-        };
-        if execution_result.is_ok() {
-            // This limit is only present in Version 3 and up, so use this to gate it
-            if let (Some(normal_lim), Some(system_lim)) = (
-                protocol_config.max_size_written_objects_as_option(),
-                protocol_config.max_size_written_objects_system_tx_as_option(),
-            ) {
-                let written_objects_size = temporary_store.written_objects_size();
-
-                match check_limit_by_meter!(
-                    !gas_charger.is_unmetered(),
-                    written_objects_size,
-                    normal_lim,
-                    system_lim,
-                    metrics.excessive_written_objects_size
-                ) {
-                    LimitThresholdCrossed::None => (),
-                    LimitThresholdCrossed::Soft(_, limit) => {
-                        warn!(
-                            written_objects_size = written_objects_size,
-                            soft_limit = limit,
-                            "Written objects size crossed soft limit",
-                        )
-                    }
-                    LimitThresholdCrossed::Hard(_, lim) => {
-                        execution_result = Err(ExecutionError::new_with_source(
-                            ExecutionErrorKind::WrittenObjectsTooLarge {
-                                current_size: written_objects_size as u64,
-                                max_size: lim as u64,
-                            },
-                            "Written objects size crossed hard limit",
-                        ))
-                    }
-                };
+                metrics,
+            );
+            if let Err(e) = gas_check {
+                execution_result = Err(e);
             }
         }
 
         execution_result
     });
 
     let cost_summary = gas_charger.charge_gas(temporary_store, &mut result);
+
+    if let Err(e) = run_conservation_checks::<Mode>(
+        temporary_store,
+        gas_charger,
+        tx_ctx,
+        move_vm,
+        enable_expensive_checks,
+        &cost_summary,
+        is_genesis_tx,
+        advance_epoch_gas_summary,
+    ) {
+        result = Err(e);
+    }
+
+    (cost_summary, result)
+}
+
+#[instrument(name = "run_conservation_checks", level = "debug", skip_all)]
```

Comment:
I'm not really sure what this does. Do we need instrumentation for these functions?

Reply:
this is for the tracing, it would help if using the logs to debug something